libpod/networking_linux.go

```diff
@@ -162,19 +182,92 @@ func (r *Runtime) setupRootlessNetNS(ctr *Container) (err error) {
 			if os.IsTimeout(err) {
 				// Check if the process is still running.
 				var status syscall.WaitStatus
-				_, err := syscall.Wait4(cmd.Process.Pid, &status, syscall.WNOHANG, nil)
+				pid, err := syscall.Wait4(cmd.Process.Pid, &status, syscall.WNOHANG, nil)
 				if err != nil {
 					return errors.Wrapf(err, "failed to read slirp4netns process status")
 				}
+				if pid != cmd.Process.Pid {
+					continue
+				}
 				if status.Exited() || status.Signaled() {
 					return errors.New("slirp4netns failed")
 				}
-
 				continue
 			}
 			return errors.Wrapf(err, "failed to read from slirp4netns sync pipe")
 		}
 	}
+
+	if havePortMapping {
+		const pidWaitTimeout = 60000
```

Comment:
Can we get a unit here? 60000 seems like a bit number, not knowing if it's seconds/milliseconds/nanoseconds? Comment describing it would be fine

Reply:
updated